CMakeLists.txt

```diff
@@ -53,20 +79,32 @@ if(ENABLE_LTO)
     include(CheckIPOSupported)
     check_ipo_supported(RESULT ipo_supported OUTPUT ipo_error)
 
-    if(ipo_supported AND (CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "MinSizeRel"))
-        message(STATUS "IPO / LTO enabled")
-        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
-    elseif(ipo_supported)
-        message(STATUS "Not enabling IPO / LTO on debug builds")
+    if(ipo_supported)
+        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_RELEASE TRUE)
+        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION_MINSIZEREL TRUE)
+        if(CMAKE_BUILD_TYPE)
+            if(CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "MinSizeRel")
+                message(STATUS "IPO / LTO enabled")
+            else()
+                message(STATUS "Not enabling IPO / LTO on debug builds")
+            endif()
+        else()
+            message(STATUS "IPO / LTO will only be enabled for release builds")
+        endif()
     else()
         message(STATUS "IPO / LTO not supported: <${ipo_error}>")
     endif()
 endif()
 
 option(BUILD_TESTING "Build the testing tree." ON)
 
-find_package(ECM REQUIRED NO_MODULE)
-set(CMAKE_MODULE_PATH "${ECM_MODULE_PATH};${CMAKE_MODULE_PATH}")
+find_package(ECM QUIET NO_MODULE)
+if(NOT ECM_FOUND)
+    message(STATUS "Using bundled ECM")
+    set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/libraries/extra-cmake-modules/modules;${CMAKE_MODULE_PATH}")
```

Comment:
I don't think I like this, as *some* functions will not be available this way. But there is no better way currently :|

Reply:
This might fail silently and cause errors later. Maybe check if the local clone of ECM even exists (i.e. if user forgot to clone submodules)

Reply:
Yeah, it's not really the intended way to use ECM, aside from compiling it before hand we could
Use vcpkg or conan.io package managers, but I don't know how widely used either of those are, at least vcpkg has a ready to go action
Or
Use CMakes's `execute_process()` function to build ECM during PrismLsuncher's configuration stage.

I should be able to add a check for an empty ECM folder later today